packages/neon/neon/lib/src/models/push_notification.dart

```diff
@@ -4,25 +4,56 @@ import 'package:nextcloud/nextcloud.dart';
 
 part 'push_notification.g.dart';
 
+/// The json key for [PushNotification.subject].
+const String _subjectKey = 'subject';
+
+/// Data for a push notification.
 @JsonSerializable()
 @immutable
 @internal
 class PushNotification {
+  /// Creates a new push notification.
   const PushNotification({
     required this.accountID,
     required this.priority,
     required this.type,
     required this.subject,
   });
 
-  factory PushNotification.fromJson(final Map<String, dynamic> json) => _$PushNotificationFromJson(json);
+  /// Creates a new PushNotification object from the given [json] data.
+  ///
+  /// Optionally decrypts the subject if [privateKey] is supplied.
+  factory PushNotification.fromJson(
+    final Map<String, dynamic> json, [
+    final RSAPrivateKey? privateKey,
+  ]) {
+    if (privateKey != null) {
+      final encrypted = json[_subjectKey];
+      if (encrypted is! String) {
+        throw StateError(
+          'Expected a String but got ${encrypted.runtimeType}. Subject must be a String if a privateKey is supplied.',
+        );
+      }
+
+      json[_subjectKey] = decryptPushNotificationSubject(privateKey, encrypted);
+    }
+
+    return _$PushNotificationFromJson(json);
```

Comment:
Now if the key is not supplied the original value is kept. Parsing will now fail because it expects a json object and not a string. I don't really like this approach you took here, the old one was better imo

Reply:
Please add a unit test for this if you want to continue this way

Reply:
My idea was to get rid of the custom parsing in `push_utils.dart` I'm open to other ways to achieve this.

Reply:
I'll drop it for this PR